WordPress/Classes/Services/AccountSettingsService.swift

```diff
@@ -0,0 +1,134 @@
+            // TODO: show/return error to the user (@koke 2015-11-24)
+            // What should be showing the error? Let's post a notification for now so something else can handle it
+            NSNotificationCenter.defaultCenter().postNotificationName(AccountSettingsServiceChangeSaveFailedNotification, object: error as NSError)
+        }
+    }
+
+    private func applyChange(change: AccountSettingsChange) throws -> AccountSettingsChange {
+        guard let settings = accountSettingsWithID(accountID) else {
+            DDLogSwift.logError("Tried to apply a change to nonexistent settings (ID: \(accountID)")
+            throw Errors.NotFound
+        }
+
+        let reverse = settings.applyChange(change)
+
+        ContextManager.sharedInstance().saveContext(context)
+
+        return reverse
+    }
+
+    private func updateSettings(settings: AccountSettings) {
+        if let managedSettings = accountSettingsWithID(accountID) {
+            managedSettings.updateWith(settings)
+        } else {
+            createAccountSettings(accountID, settings: settings)
+        }
+
+        ContextManager.sharedInstance().saveContext(context)
+    }
+
+    private func accountSettingsWithID(accountID: Int) -> ManagedAccountSettings? {
+        let request = NSFetchRequest(entityName: ManagedAccountSettings.entityName)
+        request.predicate = NSPredicate(format: "account.userID = %d", accountID)
+        request.fetchLimit = 1
+        let results = (try? context.executeFetchRequest(request) as! [ManagedAccountSettings]) ?? []
+        return results.first
+    }
+
+    private func createAccountSettings(accountID: Int, settings: AccountSettings) {
+        let accountService = AccountService(managedObjectContext: context)
+        guard let account = accountService.findAccountWithUserID(accountID) else {
+            DDLogSwift.logError("Tried to create settings for a missing account (ID: \(accountID)): \(settings)")
+            return
+        }
+
+        let managedSettings = NSEntityDescription.insertNewObjectForEntityForName(ManagedAccountSettings.entityName, inManagedObjectContext: context) as! ManagedAccountSettings
+        managedSettings.updateWith(settings)
+        managedSettings.account = account
+    }
+
+    enum Errors: ErrorType {
+        case NotFound
+    }
+}
+
+class AccountSettingsSubscription {
+    private var subscription: NSObjectProtocol? = nil
+
+    init(accountID: Int, context: NSManagedObjectContext, changed: ManagedAccountSettings? -> Void) {
+        self.subscription = NSNotificationCenter.defaultCenter().addObserverForName(NSManagedObjectContextDidSaveNotification, object: context, queue: NSOperationQueue.mainQueue()) { (notification) -> Void in
+            // FIXME: Inspect changed objects in notification instead of fetching for performance (@koke 2015-11-23)
+            let account = self.fetchAccount(accountID, context: context)
```

Comment:
But that would mean keeping a reference to the managed object, which I was trying to avoid. And account can be `nil` at some point